mlflow/models/utils.py

```diff
@@ -146,10 +156,24 @@ def _read_example(mlflow_model: Model, path: str):
     if mlflow_model.saved_input_example_info is None:
         return None
     example_type = mlflow_model.saved_input_example_info["type"]
-    if example_type != "dataframe":
+    if example_type not in ["dataframe", "ndarray"]:
         raise MlflowException(
             "This version of mlflow can not load example of type {}".format(example_type)
         )
     input_schema = mlflow_model.signature.inputs if mlflow_model.signature is not None else None
     path = os.path.join(path, mlflow_model.saved_input_example_info["artifact_path"])
-    return _dataframe_from_json(path, schema=input_schema, precise_float=True)
+    if (example_type == "ndarray") or (input_schema and input_schema.is_tensor_spec()):
+        return _read_tensor_input_from_json(path)
+    else:
+        return _dataframe_from_json(path, schema=input_schema, precise_float=True)
+
+
+def _read_tensor_input_from_json(path):
+    with open(path, "r") as handle:
+        try:
+            inp_dict = json.load(handle)
+        except Exception:
```

Comment:
Not sure if we need to catch&rethrow, it may be better to just throw the underlying error.